src/altscore/borrower_central/model/deals.py

```
@@ -403,3 +426,26 @@ async def query_by_status(self, status: str, page: int = 1, per_page: int = 10):
             )
             await raise_for_status_improved(response)
             return response.json()
+
+    @retry_on_401_async
+    async def retrieve_by_external_id(self, external_id: str) -> Optional[DealAsync]:
+        """
+        Retrieve a deal by its external ID
+        """
+        async with httpx.AsyncClient(base_url=self.altscore_client._borrower_central_base_url) as client:
+            response = await client.get(
+                f"/v1/deals",
```

Ruff (0.14.0)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)